add cli-commands for db-administration

pmattmann opened the pull request:
Commands:
- Rebuild database schema (drop & create)
  - vendor/bin/laminas rebuild-database
- Update database schema (migration)
  - vendor/bin/laminas update-database
- Load Prod Data-Fixtures
  - vendor/bin/laminas load-data-fixture --path=module/\*/data/prod
- Load all Data-Fixtures
  - vendor/bin/laminas load-data-fixture --path=module/\*/data/\*

pmattmann committed:
add cli-commands for db-administration

carlobeltrame committed:
Clear and reseed database when deploying

pmattmann committed:
composer update

pmattmann committed:
Merge branch 'devel' into fix/di-generate-aot

BacLuc commented on .docker-hub/backend/docker-entrypoint.sh:
```diff
@@ -1,6 +1,7 @@
 php bin/wait-for-db.php
-vendor/bin/doctrine orm:schema-tool:update --force --complete
+vendor/bin/laminas rebuild-database
+vendor/bin/laminas load-data-fixture --path=module/*/data/*
```
hier würde der module/eCampCore/data/* reichen.
Die anderen Module sollten bis jetzt noch keine persistenz haben, oder?
Und müsste man sich hier nicht entscheiden, ob man dev oder prod daten möchte?

carlobeltrame replied:
Zu den Modulen habe ich den Befehl von Forte übernommen. @pmattmann was meinst du, ist es plausibel dass in Zukunft andere Module Persistenz dazubekommen und das hier dann vergessen geht? Falls nicht ändere ich das gerne ab.

Und Dev oder Prod Daten, auch ein guter Punkt, aber auf einer wirklich produktiven Produktion müsste man diesen Mechanismus dann sowieso ausschalten, sonst werden jedes Mal beim Deployen die Daten der User gespült. Ich habe noch zwei Notizen ins [Release-Projekt](https://github.com/ecamp/ecamp3/projects/2) gemacht, sodass das dann nicht vergessen geht. Zu bedenken ist, dass das Zurücksetzen aktuell fix in die Docker containers auf Docker Hub eingebacken ist, aber ich schätze das Risiko klein ein dass jemand vor unserem offiziellen Release schon diese Container verwendet und sich aus Versehen die Datenbank damit leert.

pmattmann replied:
Aus dem Verzeichnis ```module``` erwarte ich aktuell keine anderen Fixtures. Daher könnte man das auch fixieren...
Eher könnte es passieren, dass ein ContentType mal eigene Fixtures mit sich bring. Dann bedarf es eine zusätzliche Ziele (da anderes Verzeichnis).

carlobeltrame committed:
Be more specific about where to get fixtures.

This will need to be changed when the first ContentType with Fixtures is added.

carlobeltrame reviewed (changes requested):
Ich schreibe noch Tests für die Commands. Abgesehen davon passts für mich.

carlobeltrame committed:
Add tests for the new console commands

carlobeltrame committed:
Rename command since it loads all PHP files in the given directory paths

carlobeltrame committed:
Configure DI correctly so Laminas can resolve all dependencies for the commands

carlobeltrame committed:
CS fix
